static/app/utils/discover/eventView.tsx

```diff
@@ -1386,7 +1393,30 @@ class EventView {
     });
     return conditions.formatString();
   }
+
+  /**
+   * Eventview usually just holds onto a project id for selected projects.
+   * Sometimes we need to iterate over the related project objects, this will give you the full projects if the Projects list is passed in.
+   * Also covers the 'My Projects' case which is sometimes missed, tries using the 'isMember' property of projects to pick the right list.
+   */
+  getFullSelectedProjects(fullProjectList: Project[]) {
+    const selectedProjectIds = this.project;
+    const isMyProjects = selectedProjectIds.length === 0;
+    if (isMyProjects) {
+      return fullProjectList.filter(p => p.isMember);
+    }
+    const isAllProjects =
+      selectedProjectIds.length === 1 && selectedProjectIds[0] === ALL_ACCESS_PROJECTS;
+    if (isAllProjects) {
+      return fullProjectList;
+    }
+
+    const projectMap = Object.fromEntries(fullProjectList.map(p => [String(p.id), p]));
+
+    return selectedProjectIds.map(id => projectMap[String(id)]);
+  }
 }
+``;
```

Comment:
@k-fish I believe you added this by mistake

Reply:
I addressed this in the commit https://github.com/getsentry/sentry/pull/39510/commits/23ebe94461e6a991a7874f2d97ca1c2cdd11875d